lib/ansible/galaxy/collection/concrete_artifact_manager.py

```diff
@@ -440,6 +444,10 @@ def _extract_collection_from_git(repo_url, coll_ver, b_path):
 
 
 # FIXME: use random subdirs while preserving the file names
+@retry_with_delays_and_condition(
+    backoff_iterator=functools.partial(generate_jittered_backoff, retries=6, delay_base=2, delay_threshold=40),
+    should_retry_error=should_retry_error
+)
 def _download_file(url, b_path, expected_hash, validate_certs, token=None, timeout=60):
```

Comment:
Note to self: This method needs updated to return `AnsibleError` with `orig_exc`.